interfaces/builtin/ofono.go

```diff
@@ -0,0 +1,332 @@
+/dev/net/tun rw,
+# to create ppp network interfaces
+capability net_admin,
+
+# To check present devices
+/run/udev/data/+usb:* r,
+/run/udev/data/+pci:* r,
+/run/udev/data/c* r,
+/run/udev/data/n* r,
+/sys/bus/usb/devices/ r,
+# FIXME snapd should be querying udev and adding the /sys and /run/udev accesses
+# that are assigned to the snap, but we are not there yet.
+/sys/bus/usb/devices/** r,
+
+# To get current seat, used to know user preferences like default SIM in
+# multi-SIM devices.
+/run/systemd/seats/{,*} r,
+
+# Access to modem ports
+# FIXME snapd should be more dynamic to avoid conflicts between snaps trying to
+# access same ports.
+/dev/tty[^0-9]* rw,
+/dev/cdc-* rw,
+
+network netlink raw,
+network netlink dgram,
+network bridge,
+network inet,
+network inet6,
+network packet,
+network bluetooth,
+
+include <abstractions/nameservice>
+
+# DBus accesses
+include <abstractions/dbus-strict>
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member={Request,Release}Name
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+# Allow binding the service to the requested connection name
+dbus (bind)
+    bus=system
+    name="org.ofono",
+
+# Allow traffic to/from our path and interface with any method
+dbus (receive, send)
+    bus=system
+    path=/{,**}
+    interface=org.ofono.*,
+`
+
+const ofonoConnectedSlotAppArmor = `
+# Allow service to interact with connected clients
+
+# Allow traffic to/from our interfaces. The path depends on the modem plugin,
+# and is arbitrary.
+dbus (receive, send)
+    bus=system
+    path=/{,**}
+    interface=org.ofono.*
+    peer=(label=###PLUG_SECURITY_TAGS###),
+`
+
+const ofonoConnectedPlugAppArmor = `
+# Description: Allow using Ofono service. Reserved because this gives
+#  privileged access to the Ofono service.
+# Usage: reserved
+
+#include <abstractions/dbus-strict>
+
+# Allow all access to ofono services
+dbus (receive, send)
+    bus=system
+    path=/{,**}
+    interface=org.ofono.*
+    peer=(label=###SLOT_SECURITY_TAGS###),
+`
+
+const ofonoConnectedPlugAppArmorClassic = `
+# Allow all access to ofono services on classic
+dbus (receive, send)
+    bus=system
+    path=/{,**}
+    interface=org.ofono.*
+    peer=(label=unconfined),
+`
+
+const ofonoPermanentSlotSecComp = `
+# Description: Allow operating as the ofono service. Reserved because this
+#  gives privileged access to the system.
+# Usage: reserved
+
+# Communicate with DBus, netlink, rild
+accept
+accept4
+bind
+connect
+getpeername
+getsockname
+getsockopt
+listen
+recv
+recvfrom
+recvmmsg
+recvmsg
+send
+sendmmsg
+sendmsg
+sendto
+setsockopt
+shutdown
+socketpair
+socket
+`
+
+const ofonoConnectedPlugSecComp = `
+# Description: Allow using ofono service. Reserved because this gives
+#  privileged access to the ofono service.
+# Usage: reserved
+
+# Can communicate with DBus system service
+connect
+getsockname
+recv
+recvmsg
+recvfrom
+send
+sendto
+sendmsg
+socket
```

Comment:
Please remove: connect, getsockname and socket. They are all included in the default template.